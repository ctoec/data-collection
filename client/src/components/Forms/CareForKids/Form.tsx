import React, { useState, useContext } from 'react';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { CareForKidsField } from './CareForKidsField';
import { EditFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
/*
 * Basic functional component designed to allow user to edit
 * the Care For Kids field of a Child object.
 */
export const CareForKidsForm: React.FC<EditFormProps> = ({
  child,
  onSuccess,
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [isSaving, setIsSaving] = useState(false);

  if (!child) return <></>;

  const onSubmit = (updatedChild: Child) => {
    setIsSaving(true);
    apiPut(`children/${child.id}`, updatedChild, { accessToken })
      .then(() => onSuccess())
      .catch((err) => {
        console.log(err);
      })
      .finally(() => setIsSaving(false));
  };

  return (
    <div className="grid-container margin-top-2">
      <h2 className="grid-row">Care 4 Kids</h2>
      <div>
        <Form<Child>
          className="CareForKidsForm"
          data={child}
          onSubmit={onSubmit}
          noValidate
          autoComplete="off"
        >
          <CareForKidsField />
          <div className="grid-row margin-top-2">
            <FormSubmitButton text={isSaving ? 'Saving...' : 'Save'} />
          </div>
        </Form>
      </div>
    </div>
  );
};
