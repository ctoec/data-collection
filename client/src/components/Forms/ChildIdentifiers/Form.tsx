import React, { useContext, useEffect, useState } from 'react';
import {
  SasidField,
  FirstNameField,
  MiddleNameField,
  LastNameField,
  SuffixField,
  DateOfBirthField,
  BirthCertificateFieldSet,
} from './Fields';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { RecordFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child, UniqueIdType } from '../../../shared/models';
import { apiPost, apiPut } from '../../../utils/api';
import useIsMounted from '../../../hooks/useIsMounted';
import { useValidationErrors } from '../../../hooks/useValidationErrors';
import { getValidationStatusForFields } from '../../../utils/getValidationStatus';
import { useHistory } from 'react-router-dom';
import { UniqueIdField } from './Fields/UniqueId';

// The fields we use to check to see if this form has errors or missing info
const childIdentifiersFields = [
  'firstName',
  'lastName',
  'birthdate',
  'birthCertificateType',
  'birthTown',
  'birthState',
  'birthCertificateId',
];
export const doesChildIdFormHaveErrors = (child?: Child) =>
  child ? !!getValidationStatusForFields(child, childIdentifiersFields) : true;

export const ChildIdentifiersForm = ({
  child: inputChild,
  afterSaveSuccess,
  hideHeader = false,
  hideErrors,
  showFieldOrFieldset = () => true,
  setAlerts,
}: RecordFormProps) => {
  const { accessToken } = useContext(AuthenticationContext);
  const history = useHistory();
  const isMounted = useIsMounted();
  const [saving, setSaving] = useState(false);

  if (!inputChild) {
    throw new Error('Child info rendered without child');
  }
  // Must keep track of child locally in case creation fails
  const [child, updateChild] = useState(inputChild);
  useEffect(() => {
    updateChild(inputChild);
  }, [inputChild]);
  const { errorsHidden } = useValidationErrors(hideErrors);

  const onFinally = () => {
    if (isMounted()) {
      setSaving(false);
    }
  };

  const onFormSubmit = (_child: Child) => {
    setSaving(true);
    if (!child.id) {
      apiPost('children', _child, {
        accessToken,
      })
        .then((res) => {
          history.replace({ pathname: `/create-record/${res.id}` });
          afterSaveSuccess();
        })
        .catch((err) => {
          if (err.data) {
            updateChild(err.data);
          }
          console.error(err);
          setAlerts([
            { type: 'error', text: 'Unable to save child identifiers' },
          ]);
        })
        .finally(onFinally);
    } else {
      apiPut(`children/${child.id}`, _child, { accessToken })
        .then(afterSaveSuccess)
        .catch((err) => {
          console.error(err);
          setAlerts([
            { type: 'error', text: 'Unable to save child identifiers' },
          ]);
        })
        .finally(onFinally);
    }
  };

  return (
    <Form<Child>
      className="ChildIdentifiersForm usa-form"
      data={child}
      onSubmit={onFormSubmit}
      noValidate
      autoComplete="off"
      hideStatus={errorsHidden}
    >
      {!hideHeader && <h2>Child's identifiers</h2>}
      {showFieldOrFieldset(child, ['sasid', 'uniqueId']) &&
        (child.organization.uniqueIdType === UniqueIdType.SASID ? (
          <div className="mobile-lg:grid-col-12">
            <SasidField />
          </div>
        ) : child.organization.uniqueIdType === UniqueIdType.Other ? (
          <div className="mobile-lg:grid-col-12">
            <UniqueIdField />
          </div>
        ) : (
          <></>
        ))}
      {showFieldOrFieldset(child, [
        'firstName',
        'lastName',
        'middleName',
        'suffix',
      ]) && (
        <>
          <div className="mobile-lg:grid-col-9">
            <FirstNameField />
          </div>
          <div className="mobile-lg:grid-col-9">
            <MiddleNameField />
          </div>
          <div className="display-flex flex-row flex-align-end grid-row grid-gap">
            <div className="mobile-lg:grid-col-9">
              <LastNameField />
            </div>
            <div className="mobile-lg:grid-col-3">
              <SuffixField />
            </div>
          </div>
        </>
      )}
      {showFieldOrFieldset(child, ['birthdate']) && <DateOfBirthField />}
      {showFieldOrFieldset(child, [
        'birthCertificateType',
        'birthCertificateId',
        'birthTown',
        'birthState',
      ]) && <BirthCertificateFieldSet />}
      <FormSubmitButton
        text={saving ? 'Saving...' : 'Save'}
        disabled={saving}
      />
    </Form>
  );
};
