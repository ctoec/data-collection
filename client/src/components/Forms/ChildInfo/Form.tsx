import React, { useContext, useState } from 'react';
import {
  RaceField,
  EthnicityField,
  GenderField,
  DisabilityServices,
  DualLanguageLearner,
  Foster,
} from './Fields';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import { RecordFormProps } from '../types';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { Child, RACE_FIELDS } from '../../../shared/models';
import { apiPut } from '../../../utils/api';
import useIsMounted from '../../../hooks/useIsMounted';
import { useValidationErrors } from '../../../hooks/useValidationErrors';
import { getValidationStatusForFields } from '../../../utils/getValidationStatus';

// The fields we use to check to see if this form has errors or missing info
const specialCircumstancesFields = [
  'receivingDisabilityServices',
  'dualLanguageLearner',
  'foster',
];
export const childInfoFields = [
  'americanIndianOrAlaskaNative',
  'asian',
  'blackOrAfricanAmerican',
  'nativeHawaiianOrPacificIslander',
  'white',
  'raceNotDisclosed',
  'hispanicOrLatinxEthnicity',
  'gender',
  ...specialCircumstancesFields,
];
export const doesChildInfoFormHaveErrors = (child?: Child) =>
  child ? !!getValidationStatusForFields(child, childInfoFields) : true;

export const ChildInfoForm = ({
  child,
  afterSaveSuccess,
  hideHeader = false,
  hideErrorsOnFirstLoad = false,
  showFieldOrFieldset = () => true,
  setAlerts,
}: RecordFormProps) => {
  const { accessToken } = useContext(AuthenticationContext);
  const isMounted = useIsMounted();
  const [saving, setSaving] = useState(false);

  if (!child) {
    throw new Error('Child info rendered without child');
  }

  console.log(child)

  const { errorsHidden, setErrorsHidden } = useValidationErrors(
    hideErrorsOnFirstLoad
  );

  const onFinally = () => {
    if (isMounted()) {
      setErrorsHidden(false);
      setSaving(false);
    }
  };

  const onFormSubmit = (_child: Child) => {
    setSaving(true);
    apiPut(`children/${child.id}`, _child, { accessToken })
      .then(afterSaveSuccess)
      .catch((err) => {
        console.error(err);
        setAlerts([
          {
            type: 'error',
            text: 'Unable to save child info',
          },
        ]);
      })
      .finally(onFinally);
  };

  return (
    <Form<Child>
      className="ChildInfoForm usa-form"
      data={child}
      onSubmit={onFormSubmit}
      noValidate
      autoComplete="off"
      hideStatus={errorsHidden}
    >
      {!hideHeader && <h2>Child info</h2>}
      {showFieldOrFieldset(child, [
        ...RACE_FIELDS,
        'hispanixOrLatinxEthnicity',
      ]) && (
          <>
            <RaceField />
            <EthnicityField />
          </>
        )}
      {showFieldOrFieldset(child, ['gender']) && <GenderField />}
      <br />

      {showFieldOrFieldset(child, specialCircumstancesFields) && (
        <h3 className="margin-bottom-0">Special circumstances</h3>
      )}
      {showFieldOrFieldset(child, ['receivesDisabilityServices']) && (
        <DisabilityServices />
      )}
      {showFieldOrFieldset(child, ['dualLanguageLearner']) && (
        <DualLanguageLearner />
      )}
      {showFieldOrFieldset(child, ['foster']) && <Foster />}
      <FormSubmitButton
        text={saving ? 'Saving...' : 'Save'}
        disabled={saving}
      />
    </Form>
  );
};
