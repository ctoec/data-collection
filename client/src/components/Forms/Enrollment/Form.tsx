import React, { useContext, useState } from 'react';
import { Alert, Form, FormSubmitButton } from '@ctoec/component-library';
import { Enrollment, Child } from '../../../shared/models';
import {
  SiteField,
  EnrollmentStartDateField,
  CareModelField,
  AgeGroupField,
  NewFundingField,
} from './Fields';
import UserContext from '../../../contexts/UserContext/UserContext';
import { RecordFormProps } from '../types';
import { getValidationStatusForFields } from '../../../utils/getValidationStatus';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { getCurrentEnrollment } from '../../../utils/models';
import { apiPost, apiPut } from '../../../utils/api';
import useIsMounted from '../../../hooks/useIsMounted';
import { useValidationErrors } from '../../../hooks/useValidationErrors';

const enrollmentFields = ['site', 'ageGroup', 'entry', 'model'];
const enrollmentFundingFields = [...enrollmentFields, 'fundings'];
export const doesEnrollmentFormHaveErrors = (
  child?: Child,
  enrollmentId?: number,
  opts: { excludeFundings?: boolean } = {}
) => {
  if (enrollmentId) {
    const enrollment = child?.enrollments?.find((e) => e.id === enrollmentId);
    if (enrollment && !enrollment.fundings?.length) return true;
    return enrollment
      ? !!getValidationStatusForFields(
        enrollment,
        opts.excludeFundings ? enrollmentFields : enrollmentFundingFields
      )
      : false;
  }

  if (child && !!getValidationStatusForFields(child, ['enrollments'])) {
    return true;
  }

  return child?.enrollments?.length
    ? !!getValidationStatusForFields(child.enrollments, enrollmentFundingFields)
    : true;
};

type EnrollmentFormProps = {
  id?: string;
  enrollmentId?: number;
} & RecordFormProps;

export const EnrollmentForm: React.FC<EnrollmentFormProps> = ({
  id,
  child,
  enrollmentId,
  AdditionalButton,
  afterSaveSuccess,
  setAlerts,
  showFieldOrFieldset = () => true,
  hideErrorsOnFirstLoad,
}) => {
  if (!child) {
    throw new Error('Enrollment form rendered without child');
  }

  const { errorsHidden, setErrorsHidden } = useValidationErrors(
    hideErrorsOnFirstLoad
  );

  const [loading, setLoading] = useState(false);
  const { accessToken } = useContext(AuthenticationContext);

  const { user } = useContext(UserContext);
  const sites = user?.sites || [];

  const enrollment =
    (enrollmentId
      ? child.enrollments?.find((e) => e.id === enrollmentId)
      : getCurrentEnrollment(child)) || ({} as Enrollment);

  const createEnrollment = async (updatedData: Enrollment) =>
    apiPost(
      `children/${child.id}/change-enrollment`,
      { newEnrollment: updatedData },
      { accessToken, jsonParse: false }
    );

  const updateEnrollment = async (updatedData: Enrollment) =>
    apiPut(`enrollments/${enrollment.id}`, updatedData, {
      accessToken,
    });

  const saveData = enrollment.id ? updateEnrollment : createEnrollment;

  const isMounted = useIsMounted();

  const onFinally = () => {
    if (isMounted()) {
      setErrorsHidden(false);
      setLoading(false);
    }
  };
  const onSubmit = (updatedData: Enrollment) => {
    setLoading(true);

    //  Because the radio group for the Site field broadcasts ID as a string,
    //  but the API is expecting a number (as it should)
    if (
      !!updatedData.site &&
      !!updatedData.site.id &&
      typeof updatedData.site.id === 'string'
    ) {
      updatedData.site.id = parseInt(updatedData.site.id);
    }
    saveData(updatedData)
      .then(afterSaveSuccess)
      .catch((err) => {
        console.error(err);
        setAlerts([
          {
            type: 'error',
            text: 'Unable to save enrollment',
          },
        ]);
      })
      .finally(onFinally);
  };

  return (
    <>
      {!enrollment.id && (
        <Alert
          type="info"
          text="Adding a child who has withdrawn, changed sites or age groups? Add their earliest enrollment then make changes from the child's record in your roster."
        />
      )}
      <Form<Enrollment>
        id={id || `edit-enrollment-form-${enrollment.id}`}
        data={enrollment}
        onSubmit={onSubmit}
        hideStatus={errorsHidden}
      >
        {showFieldOrFieldset(enrollment, ['site']) && (
          <SiteField<Enrollment>
            sites={sites.filter(
              (s) => s.organizationId === child?.organization?.id
            )}
          />
        )}
        {showFieldOrFieldset(enrollment, ['entry', 'model']) && (
          <>
            <EnrollmentStartDateField<Enrollment> />
            <CareModelField<Enrollment> />
          </>
        )}
        {showFieldOrFieldset(enrollment, ['ageGroup']) && (
          <AgeGroupField<Enrollment> />
        )}
        {showFieldOrFieldset(enrollment, ['fundings']) && (
          <NewFundingField<Enrollment>
            fundingAccessor={(data) => data.at('fundings').at(0)}
            getEnrollment={(data) => data.value}
            organizationId={child?.organization?.id}
          />
        )}
        {AdditionalButton}
        <FormSubmitButton
          text={loading ? 'Saving...' : 'Save'}
          disabled={loading}
        />
      </Form>
    </>
  );
};
