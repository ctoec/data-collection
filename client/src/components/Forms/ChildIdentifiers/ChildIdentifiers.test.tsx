import React from 'react';
import {
  snapshotTestHelper,
  accessibilityTestHelper,
} from '../../../testHelpers';
import { ChildIdentifiersForm } from './Form';
import { Family } from '../../../shared/models';

const child = {
  id: '00000000-0000-0000-0000-000000000000',
  firstName: 'First',
  lastName: 'Last',
  family: {} as Family,
};

describe('EditRecord', () => {
  describe('ChildIdentifiers', () => {
    snapshotTestHelper(
      <ChildIdentifiersForm
        child={child}
        afterDataSave={jest.fn()}
        setAlerts={jest.fn()}
      />,
      { wrapInRouter: true }
    );

    accessibilityTestHelper(
      <ChildIdentifiersForm
        child={child}
        afterDataSave={jest.fn()}
        setAlerts={jest.fn()}
      />,
      { wrapInRouter: true }
    );
  });
});
