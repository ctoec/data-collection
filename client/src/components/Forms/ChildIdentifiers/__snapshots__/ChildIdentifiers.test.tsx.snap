// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecord ChildIdentifiers matches snapshot - no unique id 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="usa-form ChildIdentifiersForm usa-form"
    novalidate=""
  >
    <h2>
      Child's identifiers
    </h2>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="firstName"
        >
          First name 
        </label>
        <input
          aria-required="true"
          class="usa-input oec-text-input"
          id="firstName"
          type="text"
          value="First"
        />
      </div>
    </div>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="middleName"
        >
          Middle name 
          <span
            class="usa-hint"
          >
             (optional)
          </span>
        </label>
        <input
          aria-required="false"
          class="usa-input oec-text-input"
          id="middleName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="display-flex flex-row flex-align-end grid-row grid-gap"
    >
      <div
        class="mobile-lg:grid-col-9"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="lastName"
          >
            Last name 
          </label>
          <input
            aria-required="true"
            class="usa-input oec-text-input"
            id="lastName"
            type="text"
            value="Last"
          />
        </div>
      </div>
      <div
        class="mobile-lg:grid-col-3"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="suffix"
          >
            Suffix 
            <span
              class="usa-hint"
            >
               (optional)
            </span>
          </label>
          <input
            aria-required="false"
            class="usa-input oec-text-input"
            id="suffix"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <fieldset
      aria-labelledby="fieldset-legend-dateOfBirth-picker dateOfBirth-picker-hint"
      class="usa-fieldset"
      id="dateOfBirth-picker"
    >
      <legend
        id="fieldset-legend-dateOfBirth-picker"
      >
        <span
          class="usa-label"
        >
          Birthdate
        </span>
      </legend>
      <span
        class="usa-hint text-italic"
        id="dateOfBirth-picker-hint"
      >
        For example: 6 21 2008
      </span>
      <div>
        <div
          class="flex-row flex-align-end usa-memorable-date"
        >
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-month"
            >
              Month 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-month"
              max="12"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-day"
            >
              Day 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-day"
              max="31"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-year"
            >
              Year 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-year"
              max="2200"
              min="1000"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      aria-labelledby="fieldset-legend-birth-certificate-fields"
      class="usa-fieldset"
      id="birth-certificate-fields"
    >
      <legend
        id="fieldset-legend-birth-certificate-fields"
      >
        <span
          class="usa-label"
        >
          Birth certificate
        </span>
      </legend>
      <div>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="US-birth-certificate"
            >
              US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Non-US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="Non-US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="Non-US-birth-certificate"
            >
              Non-US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Unavailable"
              name="birthCertificateType"
              type="radio"
              value="Unavailable"
            />
            <label
              class="usa-radio__label"
              for="Unavailable"
            >
              Unavailable
            </label>
          </div>
        </span>
      </div>
    </fieldset>
    <input
      class="usa-button"
      type="submit"
      value="Save"
    />
  </form>
</DocumentFragment>
`;

exports[`EditRecord ChildIdentifiers matches snapshot - other unique id 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="usa-form ChildIdentifiersForm usa-form"
    novalidate=""
  >
    <h2>
      Child's identifiers
    </h2>
    <div
      class="mobile-lg:grid-col-12"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="unique-id"
        >
          Unique Id 
          <span
            class="usa-hint"
          >
             (optional)
          </span>
        </label>
        <input
          aria-required="false"
          class="usa-input oec-text-input"
          id="unique-id"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="firstName"
        >
          First name 
        </label>
        <input
          aria-required="true"
          class="usa-input oec-text-input"
          id="firstName"
          type="text"
          value="First"
        />
      </div>
    </div>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="middleName"
        >
          Middle name 
          <span
            class="usa-hint"
          >
             (optional)
          </span>
        </label>
        <input
          aria-required="false"
          class="usa-input oec-text-input"
          id="middleName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="display-flex flex-row flex-align-end grid-row grid-gap"
    >
      <div
        class="mobile-lg:grid-col-9"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="lastName"
          >
            Last name 
          </label>
          <input
            aria-required="true"
            class="usa-input oec-text-input"
            id="lastName"
            type="text"
            value="Last"
          />
        </div>
      </div>
      <div
        class="mobile-lg:grid-col-3"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="suffix"
          >
            Suffix 
            <span
              class="usa-hint"
            >
               (optional)
            </span>
          </label>
          <input
            aria-required="false"
            class="usa-input oec-text-input"
            id="suffix"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <fieldset
      aria-labelledby="fieldset-legend-dateOfBirth-picker dateOfBirth-picker-hint"
      class="usa-fieldset"
      id="dateOfBirth-picker"
    >
      <legend
        id="fieldset-legend-dateOfBirth-picker"
      >
        <span
          class="usa-label"
        >
          Birthdate
        </span>
      </legend>
      <span
        class="usa-hint text-italic"
        id="dateOfBirth-picker-hint"
      >
        For example: 6 21 2008
      </span>
      <div>
        <div
          class="flex-row flex-align-end usa-memorable-date"
        >
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-month"
            >
              Month 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-month"
              max="12"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-day"
            >
              Day 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-day"
              max="31"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-year"
            >
              Year 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-year"
              max="2200"
              min="1000"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      aria-labelledby="fieldset-legend-birth-certificate-fields"
      class="usa-fieldset"
      id="birth-certificate-fields"
    >
      <legend
        id="fieldset-legend-birth-certificate-fields"
      >
        <span
          class="usa-label"
        >
          Birth certificate
        </span>
      </legend>
      <div>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="US-birth-certificate"
            >
              US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Non-US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="Non-US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="Non-US-birth-certificate"
            >
              Non-US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Unavailable"
              name="birthCertificateType"
              type="radio"
              value="Unavailable"
            />
            <label
              class="usa-radio__label"
              for="Unavailable"
            >
              Unavailable
            </label>
          </div>
        </span>
      </div>
    </fieldset>
    <input
      class="usa-button"
      type="submit"
      value="Save"
    />
  </form>
</DocumentFragment>
`;

exports[`EditRecord ChildIdentifiers matches snapshot - sasid 1`] = `
<DocumentFragment>
  <form
    autocomplete="off"
    class="usa-form ChildIdentifiersForm usa-form"
    novalidate=""
  >
    <h2>
      Child's identifiers
    </h2>
    <div
      class="mobile-lg:grid-col-12"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="sasid"
        >
          SASID 
          <span
            class="usa-hint"
          >
             (optional)
          </span>
        </label>
        <input
          aria-required="false"
          class="usa-input oec-text-input"
          id="sasid"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="firstName"
        >
          First name 
        </label>
        <input
          aria-required="true"
          class="usa-input oec-text-input"
          id="firstName"
          type="text"
          value="First"
        />
      </div>
    </div>
    <div
      class="mobile-lg:grid-col-9"
    >
      <div
        class=" usa-form-group"
      >
        <label
          class="usa-label"
          for="middleName"
        >
          Middle name 
          <span
            class="usa-hint"
          >
             (optional)
          </span>
        </label>
        <input
          aria-required="false"
          class="usa-input oec-text-input"
          id="middleName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="display-flex flex-row flex-align-end grid-row grid-gap"
    >
      <div
        class="mobile-lg:grid-col-9"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="lastName"
          >
            Last name 
          </label>
          <input
            aria-required="true"
            class="usa-input oec-text-input"
            id="lastName"
            type="text"
            value="Last"
          />
        </div>
      </div>
      <div
        class="mobile-lg:grid-col-3"
      >
        <div
          class=" usa-form-group"
        >
          <label
            class="usa-label"
            for="suffix"
          >
            Suffix 
            <span
              class="usa-hint"
            >
               (optional)
            </span>
          </label>
          <input
            aria-required="false"
            class="usa-input oec-text-input"
            id="suffix"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <fieldset
      aria-labelledby="fieldset-legend-dateOfBirth-picker dateOfBirth-picker-hint"
      class="usa-fieldset"
      id="dateOfBirth-picker"
    >
      <legend
        id="fieldset-legend-dateOfBirth-picker"
      >
        <span
          class="usa-label"
        >
          Birthdate
        </span>
      </legend>
      <span
        class="usa-hint text-italic"
        id="dateOfBirth-picker-hint"
      >
        For example: 6 21 2008
      </span>
      <div>
        <div
          class="flex-row flex-align-end usa-memorable-date"
        >
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-month"
            >
              Month 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-month"
              max="12"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-day"
            >
              Day 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-day"
              max="31"
              min="1"
              type="number"
              value=""
            />
          </div>
          <div
            class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
          >
            <label
              class="usa-label"
              for="dateOfBirth-picker-year"
            >
              Year 
            </label>
            <input
              aria-required="true"
              class="usa-input oec-text-input usa-input--inline"
              id="dateOfBirth-picker-year"
              max="2200"
              min="1000"
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      aria-labelledby="fieldset-legend-birth-certificate-fields"
      class="usa-fieldset"
      id="birth-certificate-fields"
    >
      <legend
        id="fieldset-legend-birth-certificate-fields"
      >
        <span
          class="usa-label"
        >
          Birth certificate
        </span>
      </legend>
      <div>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="US-birth-certificate"
            >
              US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Non-US-birth-certificate"
              name="birthCertificateType"
              type="radio"
              value="Non-US birth certificate"
            />
            <label
              class="usa-radio__label"
              for="Non-US-birth-certificate"
            >
              Non-US birth certificate
            </label>
          </div>
        </span>
        <span>
          <div
            class="usa-radio"
          >
            <input
              class="usa-radio__input"
              id="Unavailable"
              name="birthCertificateType"
              type="radio"
              value="Unavailable"
            />
            <label
              class="usa-radio__label"
              for="Unavailable"
            >
              Unavailable
            </label>
          </div>
        </span>
      </div>
    </fieldset>
    <input
      class="usa-button"
      type="submit"
      value="Save"
    />
  </form>
</DocumentFragment>
`;
